app/scripts/lib/rpc-method-middleware/handlers/wallet-getPermissions.test.ts

```diff
@@ -0,0 +1,290 @@
+        caveats: [
+          {
+            value: {
+              foo: 'bar',
+            },
+          },
+        ],
+      },
+    }),
+  );
+  const getAccounts = jest
+    .fn()
+    .mockReturnValue(['0x1', '0x2', '0x3', '0xdeadbeef']);
+  const response: PendingJsonRpcResponse<Json> = {
+    jsonrpc: '2.0' as const,
+    id: 0,
+  };
+  const handler = (request: JsonRpcRequest<Json[]>) =>
+    getPermissionsHandler.implementation(request, response, next, end, {
+      getPermissionsForOrigin,
+      getAccounts,
+    });
+
+  return {
+    response,
+    next,
+    end,
+    getPermissionsForOrigin,
+    getAccounts,
+    handler,
+  };
+};
+
+describe('getPermissionsHandler', () => {
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  beforeEach(() => {
+    MockMultichain.getPermittedEthChainIds.mockReturnValue([]);
+  });
+
+  it('gets the permissions for the origin', async () => {
+    const { handler, getPermissionsForOrigin } = createMockedHandler();
+
+    await handler(baseRequest);
+    expect(getPermissionsForOrigin).toHaveBeenCalled();
+  });
+
+  it('returns permissions unmodified if no CAIP-25 endowment permission has been granted', async () => {
+    const { handler, getPermissionsForOrigin, response } =
+      createMockedHandler();
+
+    getPermissionsForOrigin.mockReturnValue(
+      Object.freeze({
+        otherPermission: {
+          id: '1',
+          parentCapability: 'otherPermission',
+          caveats: [
+            {
+              value: {
+                foo: 'bar',
+              },
+            },
+          ],
+        },
+      }),
+    );
+
+    await handler(baseRequest);
+    expect(response.result).toStrictEqual([
+      {
+        id: '1',
+        parentCapability: 'otherPermission',
+        caveats: [
+          {
+            value: {
+              foo: 'bar',
+            },
+          },
+        ],
+      },
+    ]);
+  });
+
+  describe('CAIP-25 endowment permissions has been granted', () => {
+    it('returns the permissions with the CAIP-25 permission removed', async () => {
+      const { handler, getAccounts, response } = createMockedHandler();
+      getAccounts.mockReturnValue([]);
+      await handler(baseRequest);
+      expect(response.result).toStrictEqual([
+        {
+          id: '2',
+          parentCapability: 'otherPermission',
+          caveats: [
+            {
+              value: {
+                foo: 'bar',
+              },
+            },
+          ],
+        },
+      ]);
+    });
+
+    it('gets the lastSelected sorted permissioned eth accounts for the origin', async () => {
+      const { handler, getAccounts } = createMockedHandler();
+      await handler(baseRequest);
+      expect(getAccounts).toHaveBeenCalledWith(true);
+    });
+
+    it('returns the permissions with an eth_accounts permission if some eth accounts are permissioned', async () => {
+      const { handler, response } = createMockedHandler();
```

Comment:
How are we setting up the handler so that "some eth accounts are provisioned"? Maybe we want to be explicit that `getAccounts` returns a non-empty array here?

``` suggestion
      const { handler, response } = createMockedHandler();
      getAccounts.mockReturnValue(['0x1', '0x2', '0x3', '0xdeadbeef']);

Reply:
https://github.com/MetaMask/metamask-extension/pull/27847/commits/fca9657a2ff25c87a6ecb920debadc9af3892cc8